### PR TITLE
Car cancellation: simplify logic when linking to BankSignering

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Refactor the logic when linking to the external BankSignering page during car cancellation

- Generate the URL first and then render a normal link to BankSignering page

## Justify why they are needed

- Before we tried to generate a link and then automatically navigate the user there -- this was error prone and might be blocked by the browser

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2927** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2927" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2927?utm_source=stack-comment).

File: apps/store/src/components/ConfirmationPage/ContractCard.tsx [hunk 1]
```diff
@@ -52,27 +49,31 @@ const PendingContractCard = ({ id, status, displayName, approveByDate }: Contrac
...
-          <Button onClick={handleClick(id)} loading={result.loading}>
-            {t('SWITCHING_ASSISTANT_BANK_SIGNERING_LINK')}
-          </Button>
+          {props.url ? (
+            <ButtonNextLink href={props.url} target="_blank">
+              {t('SWITCHING_ASSISTANT_BANK_SIGNERING_LINK')}
+            </ButtonNextLink>
+          ) : (
+            <Button onClick={handleClick(props.id)} loading={result.loading}>
+              {t('PAYMENT_CONNECT_IFRAME_LOAD_BUTTON', { ns: 'common' })}
```
Comment (alebedev) [inline apps/store/src/components/ConfirmationPage/ContractCard.tsx:74]: Should we change translation key since it's no longer payment connect only?

**Review by alebedev — APPROVED** [review]
Nice, feels less magical now

**Comment by alebedev** [comment]
@alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2927).

**Comment by alebedev** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2927) rebased this pull request as part of a merge.

**Comment by alebedev** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2927) rebased this pull request as part of a merge.

**Comment by alebedev** [comment]
@alebedev merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2927).

**Pull request merged**